R/rope.R

```diff
@@ -113,14 +114,20 @@ rope.numeric <- function(posterior, bounds = "default", ci = .90, verbose = TRUE
 #' @importFrom insight get_response get_parameters
 #' @importFrom stats sd
 #' @keywords internal
-.rope_models <- function(posterior, bounds = "default", ci = .90, verbose = TRUE) {
-  if (all(bounds == "default")) {
-    bounds <- rope_bounds(posterior)
-  } else if (!all(is.numeric(bounds)) | length(bounds) != 2) {
-    stop("`bounds` should be 'default' or a vector of 2 numeric values (e.g., c(-0.1, 0.1)).")
+.rope_models <- function(posterior, range = "default", ci = .90, verbose = TRUE) {
+  if (all(range == "default")) {
+    range <- rope_bounds(posterior)
+  } else if (!all(is.numeric(range)) | length(range) != 2) {
+    stop("`range` should be 'default' or a vector of 2 numeric values (e.g., c(-0.1, 0.1)).")
   }
-  list <- sapply(insight::get_parameters(posterior), rope, bounds = bounds, ci = ci, verbose = verbose, simplify = FALSE)
-  flatten_list(list, name = "Parameter")
+
+  list <- sapply(insight::get_parameters(posterior), rope, range = range, ci = ci, verbose = verbose, simplify = FALSE)
+  hdi_area <- do.call(rbind, lapply(list, attr, "HDI_area"))
+
+  r <- flatten_list(list, name = "Parameter")
```

Comment:
@DominiqueMakowski This is not very nice, but `flatten_list()` drops attrbiutes, and I need to copy along the information from the HDI for `equivalence_test()`, to get a more informative output for `print.equivalence_test()`.

Reply:
I understand, in the future, we might need to think about a robust and neat solving for this flatten_list thing, as it is also present in `parameters`.